### PR TITLE
adding new ec2_lookup module and termination support for provisioning

### PR DESCRIPTION
**Pull request by jarv**

### CODE REVIEW

**Comment by feanil** [comment]
looks good.

File: playbooks/library/ec2_lookup [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+author: John Jarvis
+'''
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key, aws_secret_key, or region.
+# It is assumed that their matching environment variables are set.
+
+# Return all instances that match the tag "Name: foo"
+- local_action:
+    module: ec2
```
Comment (e0d) [inline playbooks/library/ec2_lookup]: ec2_lookup?

File: playbooks/library/ec2_lookup [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    tags:
+        Name: foo
+'''
+
+import sys
+
+AWS_REGIONS = ['ap-northeast-1',
```
Comment (e0d) [inline playbooks/library/ec2_lookup:72]: Can this be imported from somewhere or is it sprinkled throughout the ec2 module?

**Comment by e0d** [comment]
A couple small comments, but LGTM.

File: playbooks/library/ec2_lookup (hunk 2 above)
Comment (jarv) [inline playbooks/library/ec2_lookup:72]: Yeah I was thinking that as well as I was copying and pasting from ec2, there should probably be an ec2_common or something in core for this.

**Pull request merged**